### DISCUSSION TITLE
how to output specific values in insert when "use" function is used

### DISCUSSION THREAD

**Discussion by 330305020** [SOCI/soci#1076] [Q&A] [answered]
My case is below, person is a struct:

person p;
p.id = 1;
p.first_name = "John";
p.second_name = "Peter";
sql << insert into A vaules(:ID, :FIRST_NAME, :SECOND_NAME), use(p);
std::string a = sql.get_last_query();

In order to trace the database operation, I would like to output to a file that what values are inserted into table A, but in string a, the message is only " insert into A vaules(:ID, :FIRST_NAME, :SECOND_NAME", and I cannot see exact values. It is not convenient for me. For example I would like to see "with value: 1, John, Peter"

Do you have some idea?

**Comment by zann1x** [comment, marked as answer]
Currently, there is no supported way to do this. See https://github.com/SOCI/soci/issues/1068.

**Comment by 330305020** [comment]
Thank you.
